### PR TITLE
Create route to validate a password reset token

### PR DESCRIPTION
**Pull request by Devessier**

### CODE REVIEW

File: packages/server/app/Controllers/Http/AuthenticationController.ts [hunk 1]
```diff
@@ -315,4 +317,39 @@ export default class AuthenticationController {
...
+        });
+        const isInvalidToken = isValidToken === false;
+        if (isInvalidToken === true) {
+            response.status(400);
+
+            return {
+                status: 'INVALID_TOKEN',
+            };
+        }
+
```
Comment (prastoin) [inline packages/server/app/Controllers/Http/AuthenticationController.ts:350]: Thoughts: user will then have to sign in using his new password ? What will do or display the client after this operation success ?

File: packages/server/app/Controllers/Http/AuthenticationController.test.ts [hunk 2]
```diff
@@ -937,3 +939,159 @@ test.group('Request password reset', (group) => {
...
+        const parsedResponseBody = ValidatePasswordResetTokenResponseBody.parse(
+            response.body,
+        );
+
+        assert.deepStrictEqual(parsedResponseBody, {
+            status: 'INVALID_TOKEN',
+        });
+    });
+
+    test('Returns error if token has expired', async (assert) => {
```
Comment (prastoin) [inline packages/server/app/Controllers/Http/AuthenticationController.test.ts:1056]: praise: great coverage !

**Review by prastoin — APPROVED** [review]
Nice work !
Leaved few comments
Still ready to be merged

File: packages/server/app/Controllers/Http/AuthenticationController.ts (hunk 1 above)
Comment (Devessier) [reply packages/server/app/Controllers/Http/AuthenticationController.ts:350]: This route will be requested at this step:

CleanShot 2022-04-04 at 09 48 55@2x

It's only meant to confirm that the token is valid. After that, we redirect the user to the form where she fills her new password.
And when the user submits her new password, we send to the backend the email, the token and the new password. And we'll authenticate her automatically if the operation succeeds!

**Pull request merged**